src/UploadHandler.cpp

```diff
@@ -0,0 +1,180 @@
+#include "UploadHandler.hpp"
+#include "Server.hpp"
+
+UploadHandler::UploadHandler()
+{
+}
+
+UploadHandler::UploadHandler(const std::string &webRoot) : _webRoot(webRoot) {};
+
+UploadHandler::~UploadHandler()
+{
+}
+
+bool UploadHandler::isHarmfulExtension(const std::string &extension)
+{
+	std::vector<std::string> executableFiles;
+	executableFiles.push_back("exe");
+	executableFiles.push_back("com");
+	executableFiles.push_back("bat");
+	executableFiles.push_back("cmd");
+	executableFiles.push_back("msi");
+	executableFiles.push_back("scr");
+	executableFiles.push_back("pif");
+	executableFiles.push_back("ps1");
+	std::vector<std::string> dynamicLibraries;
+	dynamicLibraries.push_back("dll");
+	dynamicLibraries.push_back("so");
+	std::vector<std::string> scripts;
+	scripts.push_back("js");
+	scripts.push_back("vbs");
+	scripts.push_back("py");
+	scripts.push_back("php");
+	scripts.push_back("pl");
+	scripts.push_back("sh");
+	std::vector<std::string> configurationFiles;
+	configurationFiles.push_back("ini");
+	configurationFiles.push_back("inf");
+	configurationFiles.push_back("reg");
+
+	std::vector<std::string> harmfulExtensions;
+	harmfulExtensions.insert(harmfulExtensions.end(), executableFiles.begin(), executableFiles.end());
+	harmfulExtensions.insert(harmfulExtensions.end(), dynamicLibraries.begin(), dynamicLibraries.end());
+	harmfulExtensions.insert(harmfulExtensions.end(), scripts.begin(), scripts.end());
+	harmfulExtensions.insert(harmfulExtensions.end(), configurationFiles.begin(), configurationFiles.end());
+
+	if (std::find(harmfulExtensions.begin(), harmfulExtensions.end(), extension) != harmfulExtensions.end())
+	{
+		return true;
+	}
+	return false;
+}
+
+// Other possible codes:
+// 413 Payload Too Large, 401 Unauthorized, 403 Forbidden, 500 Internal Server Error
+void UploadHandler::checkFiles(const HTTPRequest &request)
+{
+	std::vector<File> files = request.getFiles();
+	std::vector<File>::iterator it;
+
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+		if (it->headers.find("filename") == it->headers.end())
+		{
+			std::cout << "422 Unprocessable Entity (Error: file does not have a name)" << std::endl;
+			return;
+		}
+	}
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+		if (it->fileContent.empty())
+		{
+			std::cout << "422 Unprocessable Entity (Error: file is empty)" << std::endl;
+			return;
+		}
+	}
+	for (it = files.begin(); it != files.end(); ++it)
+	{
+
+		std::string filename = it->headers.find("filename")->second;
+		std::string extension = filename.substr(filename.find_last_of(".") + 1);
```

Comment:
There is a special header with "Content-type". For example, "text/plain"

Reply:
As we tested the header of the request is generated by the browser through it's extension, the file content might be harder to parse than the extension here. I would say we pull it to main in this way and then decide if  we want to change it, @552020 might also have some input.
Anyway this is an additionnal thing, kindda to mimic a protection against hacking

Reply:
Sorry, I don't understand the question/problem. Can you explain?

Reply:
Daniil was suggesting I used the filecontent header, but this header is generated by the browser upon upload depending on the extension (we uploaded a .txt file with c code to test. So I can just as well check the extension in the code, it is more reliable and readable

Reply:
I don't understand what you mean when you say:

> So I can just as well check the extension in the code, it is more reliable and readable

You are trying to find the extension of the file which is contained only in the filename, right? The file content in the header of the file in the body of the request is not giving you any information about the file extenstion right?

Reply:
> You are trying to find the extension of the file which is contained only in the filename, right? 

Yes


> The file content in the header of the file in the body of the request is not giving you any information about the file extenstion right?

It could, but I choose not to use it. I decided to use the extension in the filename